src/dotnet-ef/OperationExecutor.cs

```
@@ -0,0 +1,177 @@
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Loader;
+using Microsoft.EntityFrameworkCore.Design;
+using Microsoft.EntityFrameworkCore.Design.Internal;
+using Microsoft.EntityFrameworkCore.Internal;
+using Microsoft.EntityFrameworkCore.Migrations.Design;
+using Microsoft.EntityFrameworkCore.Scaffolding.Internal;
+using NuGet.Frameworks;
+
+namespace Microsoft.EntityFrameworkCore.Commands
+{
+    public class OperationExecutor
+    {
+        private readonly LazyRef<DbContextOperations> _contextOperations;
+        private readonly LazyRef<DatabaseOperations> _databaseOperations;
+        private readonly LazyRef<MigrationsOperations> _migrationsOperations;
+
+        public OperationExecutor([CanBeNull] string startupProject, [CanBeNull] string environment)
+        {
+            var project = Path.Combine(Directory.GetCurrentDirectory(), Project.FileName);
+
+            if (startupProject == null)
+            {
+                startupProject = project;
+            }
+            else if (!startupProject.EndsWith(Project.FileName))
+            {
+                startupProject = Path.Combine(startupProject, Project.FileName);
+            }
+
+            Reporter.Verbose.WriteLine(("Using startup project '" + startupProject + "'.").Bold().Black());
+            Reporter.Verbose.WriteLine(("Using project '" + project + "'.").Bold().Black());
+
+            var startupProjectContext = GetCompatibleProjectContext(startupProject);
+
+            Reporter.Verbose.WriteLine("Build started...".Bold().Black());
+
+            var buildCommand = Command.Create(
+                Path.Combine(Environment.GetEnvironmentVariable("DOTNET_HOME"), "bin", "dotnet" + Constants.ExeSuffix),
```

Seems like we should raise an error if `DOTNET_HOME` is not defined?